status/condition_set.go

```diff
@@ -176,52 +177,49 @@ func (r ConditionSet) SetUnknown(conditionType string) {
 
 // recomputeRootCondition marks the root condition to true if all other dependents are also true.
 func (r ConditionSet) recomputeRootCondition(conditionType string) {
-	if condition := r.findUnhappyDependent(); condition != nil {
-		// Propagate unroot dependent to root condition.
+	if conditionType == r.root {
+		return
+	}
+
+	if conditions := r.findUnhappyDependents(); len(conditions) == 0 {
```

Comment:
one more nit:
if the reason is UnhealthyDependents, I think it makes sense to call this function `findUnhealthyDependents`